### PR TITLE
Update policies in Api and products from yaml

### PR DESCRIPTION
**Pull request by NakulSeerpu**

### CODE REVIEW

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 1]
```diff
@@ -133,18 +133,26 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
         {
             if (entityId.HasValue)
             {
+                var policies = await client.GetAllPoliciesAsync(subscriptionId, entityId.Value, entityType).ConfigureAwait(false);
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:136]: Why do you fetch policies? You already have the result from the ValidationResult

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 2]
```diff
@@ -133,18 +133,26 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
...
+
                 if (rateLimitPolicy != null)
                 {
                     var rateLimitPolicyRequest = new RateLimitPolicyRequest(
-                        rateLimitPolicy.Name, entityId, entityType, rateLimitPolicy.Description, rateLimitPolicy.Calls, rateLimitPolicy.RenewalPeriod);
+                     rateLimitPolicy.Name, entityId, entityType, rateLimitPolicy.Description, rateLimitPolicy.Calls, rateLimitPolicy.RenewalPeriod);
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:141]: Why it's changed here?

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 3]
```diff
@@ -133,18 +133,26 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
...
                     switch (policiesResults.RateLimitPolicy.Status)
                     {
                         case ValidationStatus.CanBeCreated:
                             var created = await client.CreateRateLimitPolicyAsync(subscriptionId, rateLimitPolicyRequest).ConfigureAwait(false);
                             this.publishResults.Add(new GatewayAutomationResult() { ResultCode = ResultCode.RateLimitPolicyCreated, EntityId = created.Id });
                             break;
                         case ValidationStatus.CanBeUpdated:
-                            // TODO: Update RateLimitPolicy
+                            var ratelimitPolicyDoc = policies.FirstOrDefault(p => p.PolicyType == "RateLimit");
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:149]: You already have rate limit result. You already know whether you should update or not.

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 4]
```diff
@@ -168,7 +176,13 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
                                 this.publishResults.Add(new GatewayAutomationResult() { ResultCode = ResultCode.CustomPolicyCreated, EntityId = created.Id });
                                 break;
                             case ValidationStatus.CanBeUpdated:
-                                // TODO: Update CustomPolicy
+                                var customPolicydoc = policies.FirstOrDefault(p => p.PolicyType == "Custom");
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:179]: Same here. You know the result.

**Review by jaceks2106 — CHANGES_REQUESTED** [review]

File: src/Kmd.Logic.Gateway.Automation/Publish.cs (hunk 1 above)
Comment (NakulSeerpu) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs]: Removed, not required

File: src/Kmd.Logic.Gateway.Automation/Publish.cs (hunk 2 above)
Comment (NakulSeerpu) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs]: Fixed

File: src/Kmd.Logic.Gateway.Automation/Publish.cs (hunk 3 above)
Comment (NakulSeerpu) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs]: Fixed, fetching from validation result now .

File: src/Kmd.Logic.Gateway.Automation/Publish.cs (hunk 4 above)
Comment (NakulSeerpu) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs]: Fixed

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 5]
```diff
@@ -168,7 +169,12 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
                                 this.publishResults.Add(new GatewayAutomationResult() { ResultCode = ResultCode.CustomPolicyCreated, EntityId = created.Id });
                                 break;
                             case ValidationStatus.CanBeUpdated:
-                                // TODO: Update CustomPolicy
+                                foreach (var custompolicy in policiesResults.CustomPolicies)
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:172]: 1. You have nested foreach loop in another foreach, that accepts the same array....... Why?
2. Remove this for each. Just update using `customPolicyResult` object from line 159.

**Review by jaceks2106 — CHANGES_REQUESTED** [review]

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 6]
```diff
@@ -136,15 +136,16 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
                 if (rateLimitPolicy != null)
                 {
                     var rateLimitPolicyRequest = new RateLimitPolicyRequest(
-                        rateLimitPolicy.Name, entityId, entityType, rateLimitPolicy.Description, rateLimitPolicy.Calls, rateLimitPolicy.RenewalPeriod);
```
[the remark this answers is not in the document]
Comment (NakulSeerpu) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs]: Done

File: src/Kmd.Logic.Gateway.Automation/Publish.cs (hunk 5 above)
Comment (NakulSeerpu) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs]: Using customPolicyResult object now.

**Pull request merged**